### PR TITLE
EVG-19921: Introduce PatchesInput.includeHidden

### PR DESCRIPTION
**Pull request by SupaJoon**
EVG-19921

### Description
When `onlyHidden` is true then only hidden patches will be returned. If onlyHidden is not provided or false, then only non-hidden patches will be returned.

### CODE REVIEW

File: model/patch/db.go [hunk 1]
```diff
@@ -160,13 +160,15 @@ func ByPatchNameStatusesCommitQueuePaginated(ctx context.Context, opts ByPatchNa
...
 
-	if !opts.IncludeHidden {
-		match[HiddenKey] = bson.M{"$ne": true}
-	}
 	// This is only used on the user patches page when we want to filter out the commit queue
 	if opts.IncludeCommitQueue != nil && !utility.FromBoolPtr(opts.IncludeCommitQueue) {
 		match[AliasKey] = commitQueueFilter
 	}
+
+	if opts.IncludeHidden != nil && !utility.FromBoolPtr(opts.IncludeHidden) {
```
Comment (hadjri) [inline model/patch/db.go:168]: You can just do `if !utility.FromBoolPtr(opts.IncludeHidden)`

File: graphql/tests/user/queries/patches-include-hidden.graphql [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  user(userId: "testuser") {
+    includeHidden: patches(patchesInput: { includeHidden: true }) {
```
Comment (sophstad) [inline graphql/tests/user/queries/patches-include-hidden.graphql:3]: Smart test configuration!

**Review by sophstad — COMMENTED** [review]
Could you update the PR title?

**Review by sophstad — APPROVED** [review]

File: model/patch/db.go (hunk 1 above)
Comment (hadjri) [reply model/patch/db.go:168]: Ah - in that case you can just use [FromBoolTPtr](https://gist.github.com/hadjri/170a0435ad2d815ceff8426b63f8238d#file-optional-go-L35-L43)

**Review by hadjri — APPROVED** [review]
approved w nit

File: graphql/schema/types/patch.graphql [hunk 3]
```diff
@@ -15,6 +15,7 @@ input PatchesInput {
   includeCommitQueue: Boolean
   limit: Int! = 0
   onlyCommitQueue: Boolean
+  includeHidden: Boolean
```
Comment (khelif96) [inline graphql/schema/types/patch.graphql:18]:
```suggestion
  includeHidden: Boolean = false
```

**Review by khelif96 — CHANGES_REQUESTED** [review]
Making the below suggestion will allow this change to be fully backwards compatible without breaking existing functionality.

File: graphql/schema/types/patch.graphql [hunk 4]
```diff
@@ -15,6 +15,7 @@ input PatchesInput {
   includeCommitQueue: Boolean
   limit: Int! = 0
   onlyCommitQueue: Boolean
+  includeHidden: Boolean = false
```
[the remark this answers is not in the document]
Comment (khelif96) [reply graphql/schema/types/patch.graphql:18]: I think you need to run gqlgen again.

**Review by khelif96 — APPROVED** [review]

**Pull request merged**